src/material-examples/form-field-custom-control/form-field-custom-control-example.ts

```diff
@@ -116,4 +126,32 @@ export class MyTelInput implements MatFormFieldControl<MyTel>, OnDestroy {
       this.elRef.nativeElement.querySelector('input')!.focus();
     }
   }
+
+  writeValue(tel: MyTel | null): void {
+    this.value = tel;
+  }
+
+  registerOnChange(fn: any): void {
+    this.onChange = fn;
+  }
+
+  registerOnTouched(fn: any): void {
+    this.onTouched = fn;
+  }
+
+  setDisabledState(isDisabled: boolean): void {
+    this.disabled = isDisabled;
+  }
+
+  _handleInput(): void {
+    this.onChange(this.parts.value);
+  }
+
+  _handleBlur(event: FocusEvent): void {
+    const inputs = Array.from(this.elRef.nativeElement.querySelectorAll('input'));
+
+    if (!inputs.some(input => event.relatedTarget === input)) {
```

Comment:
Isn't this guaranteed to always be `true`? The only inputs that have the listener are the ones inside the `elRef` from what I can tell.

Reply:
@crisbeto there are scenarios in which this evaluates to `false`. For example, if the user clicks an entirely different form control (which receives focus) then the blur event's `relatedTarget` will not be equal to one of the three `input` elements defined in `MyTelInput`'s template.

My rationale is that if the user is changing focus between the three input elements there's no need to mark the form control as `touched`. However, if they focus on any element other than one of the three inputs it is appropriate to mark the form control as `touched`.

Reply:
That makes sense, although I remember `relatedTarget` being flaky on some browsers. This would be a bit cleaner if you used the CDK focus monitor with subtree monitoring.

Reply:
@crisbeto thanks. The code is cleaner when relying on the CDK focus monitor.